march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py

```diff
@@ -73,31 +73,21 @@ def duration(self, duration):
     def get_interpolated_position(self, time):
         for i in range(0, len(self.interpolated_setpoints[0])):
             if self.interpolated_setpoints[0][i] > time:
-                return self.interpolated_setpoints[1][i - 1]
+                return self.interpolated_setpoints[1][i]
```

Comment:
why not use the get_interpolated_setpoint method from the trajectory class and get the value from a possible returned setpoint? seems like a useful function instead of using the same iterations.

Reply:
A good point. I am leaning towards leaving it like this, because get_interpolated_setpoint is not meant to return anything when the time is outside of the domain, while this method should return the closest interpolated setpoint in that case. Hence it will take some error catching logic to fix that, which will result in a more complicated code than now. 

What do you think?

Reply:
I get your point, i don't think it will result in a huge advantage to switch the logic to the other class. If u are leaning towards the current solution lets keep it like this.